include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2124,6 +2136,13 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::prolongate(
 
   const unsigned int n_lanes = VectorizedArrayType::size();
 
+  const bool fine_vectors_are_compatible =
+    this->vec_fine.get_partitioner()->is_globally_compatible(
+      *dst.get_partitioner());
```

Comment:
I think we should actually rewrite `Paritioner::is_globally_compatible()` so that it is hash based and does only perform communication during setup but not during this function call. This would fix this issue here.